File: cambiar_email.rb
require 'sinatra'
require 'json'
require_relative './db'

put '/cambiar-email' do
  content_type :json

  begin
    data = JSON.parse(request.body.read)
    old_email = data['oldEmail']
    new_email = data['newEmail']

    if old_email.nil? || new_email.nil?
      status 400
      return { error: 'Faltan datos' }.to_json
    end

    result = DB_CLIENT.execute("SELECT * FROM users WHERE email = '#{old_email}'")
    if result.each.empty?
      status 404
      return { error: 'Usuario no encontrado' }.to_json
    end

    DB_CLIENT.execute("UPDATE users SET email = '#{new_email}' WHERE email = '#{old_email}'").do

    status 200
    { message: 'Email actualizado' }.to_json

  rescue => e
    status 500
    { error: "Error: #{e.message}" }.to_json
  end
end

File: cambiar_email.php
<?php
header("Access-Control-Allow-Origin: *");
header("Access-Control-Allow-Headers: Content-Type");
header("Content-Type: application/json");

$data = json_decode(file_get_contents("php://input"), true);
$oldEmail = $data['oldEmail'];
$newEmail = $data['newEmail'];

// Conexión a SQL Server
$server = "auth-db.cny206g4cz8c.us-east-1.rds.amazonaws.com";
$connectionInfo = [
  "Database" => "auth",
  "UID" => "admin",
  "PWD" => "Distribuida123",
  "Encrypt" => true,
  "TrustServerCertificate" => true
];

$conn = sqlsrv_connect($server, $connectionInfo);

if (!$conn) {
  echo json_encode(["error" => "Conexión fallida"]);
  exit;
}

$sql = "UPDATE usuarios SET email = ? WHERE email = ?";
$params = [$newEmail, $oldEmail];
$stmt = sqlsrv_query($conn, $sql, $params);

if ($stmt && sqlsrv_rows_affected($stmt) > 0) {
  echo json_encode(["message" => "Email actualizado"]);
} else {
  echo json_encode(["error" => "No se pudo actualizar"]);
}

sqlsrv_close($conn);
?>
